lib/pbench/test/unit/common/test_config.py

```diff
@@ -1,68 +1,252 @@
+                        "logger_host": "logger.example.com",
+                        "logger_port": "",
+                    }
+                }
+
+        monkeypatch.setattr(configtools, "file_list", self.mock_file_list)
+        monkeypatch.setattr(ConfigParser, "__init__", mock_MockedConfigParser.__init__)
+        monkeypatch.setattr(ConfigParser, "get", MockedConfigParser.get)
+        with pytest.raises(ValueError) as exc:
+            config = PbenchConfig("pbench.cfg")
+            assert config.logger_type == "hostport"
+            assert config.logger_host == "logger.example.com"
+            assert exc == "No option 'logger_port' in section: 'logging'"
+
+    def test_logger_hostport(self, monkeypatch):
+        """verify logger_type as `hostport` with `host` and `port`"""
+
+        class mock_MockedConfigParser(MockedConfigParser):
+            def __init__(self):
+                self._sections = {
+                    "logging": {
+                        "logger_type": "hostport",
+                        "logger_host": "logger.example.com",
+                        "logger_port": "42",
+                    }
+                }
+
+        monkeypatch.setattr(configtools, "file_list", self.mock_file_list)
+        monkeypatch.setattr(ConfigParser, "__init__", mock_MockedConfigParser.__init__)
+        monkeypatch.setattr(ConfigParser, "get", MockedConfigParser.get)
+        config = PbenchConfig("pbench.cfg")
+        assert config.logger_type == "hostport"
+        assert config.logger_host == "logger.example.com"
+        assert config.logger_port == "42"
+
+    def test_no_logger_dir_option(self, monkeypatch):
+        """test no logger_dir option in config file"""
+
+        class mock_MockedConfigParser(MockedConfigParser):
+            def __init__(self):
+                self._sections = {
+                    "logging": {
+                        "logger_type": "file",
+                    }
+                }
+
+        monkeypatch.setattr(configtools, "file_list", self.mock_file_list)
+        monkeypatch.setattr(ConfigParser, "__init__", mock_MockedConfigParser.__init__)
+        monkeypatch.setattr(ConfigParser, "get", MockedConfigParser.get)
+        config = PbenchConfig("pbench.cfg")
+        assert config.log_dir is None
+
+    def test_logger_dir(self, monkeypatch):
+        """verify logger_dir option in config file"""
+
+        class mock_MockedConfigParser(MockedConfigParser):
+            def __init__(self):
+                self._sections = {
+                    "logging": {"logger_type": "file", "log_dir": "/mock/log_dir"}
+                }
+
+        monkeypatch.setattr(configtools, "file_list", self.mock_file_list)
+        monkeypatch.setattr(ConfigParser, "__init__", mock_MockedConfigParser.__init__)
+        monkeypatch.setattr(ConfigParser, "get", MockedConfigParser.get)
+        config = PbenchConfig("pbench.cfg")
+        assert config.log_using_caller_directory is False
```

Comment:
You don't have an assertion in any of the test cases where this value is expected to be `True`...should there be one?  (In `test_logger()`, perhaps?)

Reply:
yeah, maybe

Reply:
this is the default value of this opton and it is hard coded.

Reply:
OK, so `log_using_caller_directory` is only set to `True` by a subclass of `PbenchConfig`...so, in the tests in this, module it _should_ always be `False`.

But, it would probably be worth checking that in more than just this one test (e.g., it would be good to add this assertion to the others collected in `test_logger()`...which you've already done!  🎉).